### PR TITLE
tasks 1, 2, 3, 4

### PR DESCRIPTION
**Pull request by danielshulgin**

### CODE REVIEW

**Comment by danielshulgin** [comment]
reopen

File: Exercises/1-hoisting.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 'use strict';
 
-const fn = null;
+const fn = () => console.log('OK');
```
Comment (tshemsedinov) [inline Exercises/1-hoisting.js]: :question:

File: Exercises/4-count-types.js [hunk 2]
```diff
@@ -1,5 +1,12 @@
 'use strict';
 
-const countTypesInArray = null;
+const countTypesInArray = typeArray => {
+  const hash = {};
+  for (const value of typeArray) {
+    const t = typeof(value);
+    hash[t] = t in hash ? ++hash[t] : 1;
```
Comment (tshemsedinov) [inline Exercises/4-count-types.js]: See alternatives https://github.com/HowProgrammingWorks/DataTypes/pull/21#discussion_r324387734

File: Exercises/1-hoisting.js (hunk 1 above)
Comment (danielshulgin) [reply Exercises/1-hoisting.js]: fixed https://github.com/HowProgrammingWorks/DataTypes/pull/37/commits/8f9731ca46ee35aea06d699391d3c4cba1b3d974

File: Exercises/4-count-types.js (hunk 2 above)
Comment (danielshulgin) [reply Exercises/4-count-types.js]: is my decision also right?

**Pull request closed without merging**